samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -109,6 +129,12 @@ def __init__(
         self._aws_region = aws_region
         self._aws_profile = aws_profile
 
+        container_mode_enum = ContainerMode(container_mode)
+        self._warm_containers = container_mode_enum != ContainerMode.COLD
```

Comment:
I wonder if there's any unambiguous terminology for this. Currently `ContainerMode` suggests "warm", "cold" and "lazy" are separate modes. But this again suggests that both "warm" and "lazy" are "warm". It can be a confusing asymmetry.

(Just brainstorming) Maybe renaming the enums to `WARM_EAGER` and `WARM_LAZY` is clearer? Warm/cold describe the container shutdown behavior, and lazy/eager describe the container startup behavior.

Reply:
COLD is just used internally, but it is not part of the command container-mode parameter options, as actually removing this parameter means the option COLD (which is the current behavior), and this was the UX recommendation (not to add COLD as an option)

Reply:
I am not following why shouldn't we add COLD in the command line?